src/PointwiseFunctions/MathFunctions/Gaussian.hpp

```diff
@@ -72,24 +73,111 @@ class Gaussian<1, T, Frame::Inertial>
+ *  \brief 1D Gaussian \f$f = A \exp\left(-\frac{(x-x_0)^2}{w^2}\right)\f$
+ *
+ *  \details Input file options are: Amplitude, Width, and Center. The function
+ *  takes input coordinates of type `tnsr::I<T, VolumeDim, Fr>`, where `T` is
+ * e.g. `double` or `DataVector`, `Fr` is a frame (e.g. `Frame::Inertial`), and
+ * `VolumeDim` is the dimension of the spatial volume, i.e. 2 or 3. (The case of
+ * VolumeDim == 1 is handled specially by Gaussian<1, T, Frame::Inertial>.)
+ */
+template <size_t VolumeDim, typename T, typename Fr>
+class Gaussian : public MathFunction<VolumeDim, T, Fr> {
+ public:
+  struct Amplitude {
+    using type = double;
+    static constexpr OptionString help = {"The amplitude."};
+  };
+
+  struct Width {
+    using type = double;
+    static constexpr OptionString help = {"The width."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  struct Center {
+    using type = std::array<double, VolumeDim>;
+    static constexpr OptionString help = {"The center."};
+  };
+  using options = tmpl::list<Amplitude, Width, Center>;
+
+  static constexpr OptionString help = {
+      "Applies a Gaussian function to the input coordinates"};
+
+  WRAPPED_PUPable_decl_base_template(SINGLE_ARG(MathFunction<VolumeDim, T, Fr>),
+                                     Gaussian);  // NOLINT
+
+  explicit Gaussian(CkMigrateMessage* /*unused*/) noexcept {}
+
+  Gaussian(double amplitude, double width,
+           const std::array<double, VolumeDim>& center) noexcept;
+
+  Gaussian() = default;
+  ~Gaussian() override = default;
+  Gaussian(const Gaussian& /*rhs*/) = delete;
+  Gaussian& operator=(const Gaussian& /*rhs*/) = delete;
+  Gaussian(Gaussian&& /*rhs*/) noexcept = default;
+  Gaussian& operator=(Gaussian&& /*rhs*/) noexcept = default;
+
+  Scalar<T> operator()(const tnsr::I<T, VolumeDim, Fr>& x) const
+      noexcept override;
+  tnsr::i<T, VolumeDim, Fr> first_deriv(
+      const tnsr::I<T, VolumeDim, Fr>& x) const noexcept override;
+  tnsr::ii<T, VolumeDim, Fr> second_deriv(
+      const tnsr::I<T, VolumeDim, Fr>& x) const noexcept override;
+  tnsr::iii<T, VolumeDim, Fr> third_deriv(
+      const tnsr::I<T, VolumeDim, Fr>& x) const noexcept override;
+
+   double amplitude() const noexcept { return amplitude_; }
+   double inverse_width() const noexcept { return inverse_width_; }
+   std::array<double, VolumeDim> center() const noexcept {
+    return center_;
+  }
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& p) override;  // NOLINT
+
+ private:
+  double amplitude_{};
+  double inverse_width_{};
+  std::array<double, VolumeDim> center_{};
+};
+
+template <size_t VolumeDim, typename T, typename Fr>
+bool operator==(const Gaussian<VolumeDim, T, Fr>& lhs,
+                const Gaussian<VolumeDim, T, Fr>& rhs) noexcept;
+
+template <size_t VolumeDim, typename T, typename Fr>
+bool operator!=(const Gaussian<VolumeDim, T, Fr>& lhs,
+                const Gaussian<VolumeDim, T, Fr>& rhs) noexcept;
```

Comment:
I did this, but for !=, I had trouble getting the code to compile and link unless I also made != a friend function. Otherwise, I got errors about undefined or multiply defined symbols...if it's not ok for != to be a friend function, could someone please give me a solution for the alternative?